homeassistant/components/honeywell/climate.py

```diff
@@ -97,6 +98,37 @@ async def async_setup_entry(
             for device in data.devices.values()
         ]
     )
+    remove_stale_devices(hass, entry, data.devices)
+
+
+def remove_stale_devices(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    devices: dict[str, SomeComfortDevice],
+) -> None:
+    """Remove stale devices from device registry."""
+    device_registry = dr.async_get(hass)
+    device_entries = dr.async_entries_for_config_entry(
+        device_registry, config_entry.entry_id
+    )
+    all_device_ids: list = []
```

Comment:
Make this a set instead. Set comparisons are much fast than list membership test.

Reply:
Addressed in separate PR